src/particle.ts

```diff
@@ -0,0 +1,98 @@
+import {Random} from "@reside-ic/random";
+
+import type {DustModel, DustModelConstructable, DustModelInfo} from "./model";
+
+export class Particle {
+    /** The step (in time) that the particle is currently at */
+    public step: number;
+
+    /** The size of the particle state - this cannot be modified after
+     * creation
+     */
+    public readonly size: number;
+
+    private _model: DustModel;
+    private _y: number[];
+    private _yNext: number[];
+
+    /**
+     * Construct a particle
+     *
+     * @param model An initialised model - note that unlike the C++
+     * version of dust, all particles share the same initialised model
+     * as they do not run in parallel.
+     *
+     * @param step The step to start the particle at
+     *
+     * @param state Optional vector of state; if given then this is
+     * the initial state for the particle. Otherwise the state comes
+     * from the models `initial` method (see {@link DustModel})
+     */
+    constructor(model: DustModel, step: number, state?: readonly number[]) {
+        this._model = model;
+        this.step = step;
+        this.size = model.size();
+        if (state === undefined) {
+            this._y = model.initial(this.step);
+        } else {
+            this._y = state.slice();
```

Comment:
```suggestion
            this._y = [...state];
```
We usually tend to use spread operator to make shallow array (or object) copy though just a matter of preference.